src/Subscriptions/resources/admin/components/SubscriptionDetailsPage/Tabs/Overview/index.tsx

```diff
@@ -3,33 +3,39 @@ import SubscriptionStats from "./SubscriptionStats";
 import OverviewPanel from "@givewp/admin/components/OverviewPanel";
 import SubscriptionSummaryGrid from "./SubscriptionSummaryGrid";
 import { SubscriptionNotes } from "@givewp/admin/components/PrivateNotes";
-import { useDonationsBySubscription, useSubscriptionAmounts } from '@givewp/subscriptions/hooks';
+import { useSubscriptionAmounts } from '@givewp/subscriptions/hooks';
+import { getSubscriptionEmbeds } from '@givewp/subscriptions/common';
 import SubscriptionSummary from '@givewp/subscriptions/admin/components/SubscriptionDetailsPage/Tabs/Overview/SubscriptionSummary';
 import { getSubscriptionOptionsWindowData, useSubscriptionEntityRecord } from "@givewp/subscriptions/utils";
 import SubscriptionAnnualProjection from "./SubscriptionAnnualProjection";
 import styles from "./styles.module.scss";
 import { Spinner } from "@givewp/admin/components";
+import { __ } from '@wordpress/i18n';
 
 /**
+ * @unreleased updated logic to use embed relationships
  * @since 4.8.0
  */
 export default function SubscriptionDetailsPageOverviewTab() {
-    const { mode, adminUrl } = getSubscriptionOptionsWindowData();
+    const { adminUrl } = getSubscriptionOptionsWindowData();
     const { record: subscription, hasResolved: subscriptionsResolved, isResolving: subscriptionLoading } = useSubscriptionEntityRecord();
     const { intendedAmount } = useSubscriptionAmounts(subscription);
-    const { records: donations, hasResolved: donationsResolved, isResolving: donationsLoading } = useDonationsBySubscription(subscription?.id, mode);
+    const { donations } = getSubscriptionEmbeds(subscription);
 
-    if (!subscriptionsResolved || !donationsResolved || !subscription || !donations) {
-        // TODO: Add loading state
+    if (subscriptionLoading || !subscriptionsResolved) {
         return <Spinner />;
     }
 
+    if (!subscriptionsResolved) {
```

Comment:
We might need to choose the spinner or the message - or remove the` subscriptionResolved` from the conditional above.